packages/clay-slider/src/index.tsx

```diff
@@ -0,0 +1,147 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import classNames from 'classnames';
+import React, {useEffect, useRef, useState} from 'react';
+
+interface Props extends React.InputHTMLAttributes<HTMLInputElement> {
+	/**
+	 * Flag that will disable component features.
+	 */
+	disabled?: boolean;
+
+	/**
+	 * Sets maximum permitted value.
+	 */
+	max?: number;
+
+	/**
+	 * Sets minimum permitted value
+	 */
+	min?: number;
+
+	/**
+	 * Callback will always be called when the slider value changes.
+	 */
+	onValueChange: (value: number) => void;
+
+	/**
+	 * Flag to show tooltip or not.
+	 */
+	showTooltip?: boolean;
+
+	/**
+	 * Sets stepping interval.
+	 */
+	step?: number;
+
+	/**
+	 * Set tooltip position.
+	 */
+	tooltipPosition?: 'top' | 'bottom';
+
+	/**
+	 * Set the current value of the slider.
+	 */
+	value: number;
+}
+
+const THUMB_WIDTH = 24;
+
+const calcProgressWidth = (
```

Comment:
hey @pat270 i ended up keeping your calculation, i tried to get some but without much success i am keeping it for now, if anyone has something better please leave your thoughts.